From 11b5e59cdb9a16c1ce5fe027c85655d8ec6efaf2 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Fri, 12 Nov 2021 11:23:34 +0200
Subject: [PATCH] [SW-60588] gaudi2: monitor PCI RR occurrence

This patch monitors lbw and hbw error interrupt bit
of PCIE_IC_SEI_INTR_IND register and writes scratch pad
register 19 with pattern 0xABBA.

These error interrupt bits are set for lbw/hbw razwi
occurrence.

This is temporary patch for power on and needs
to be reverted once testing is done

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I1f628a06a2c60d71279093e4903ef73fb6fa9659
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 90e7cf5..44eabfa 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1227,3 +1227,16 @@
 
 	return boot_strap & PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_BTL_EN_MASK;
 }
+
+/*
+ * TODO: This is a temporary function and
+ * to be removed after GAUDI2 PO
+ */
+void check_razwi_intr(void)
+{
+	u32 reg = 0;
+
+	reg = regmap_readl(mmPCIE_WRAP_PCIE_IC_SEI_INTR_IND);
+	if (reg & PCIE_AXI_ERR_INTR_BIT)
+		regmap_writel(RAZWI_INTR_STATUS_PATTERN, mmGIC_RAZWI_STATUS_REG);
+}
diff --git a/arch/gaudi2/common/common.h b/arch/gaudi2/common/common.h
index 4c2adfb..7d9e428 100644
--- a/arch/gaudi2/common/common.h
+++ b/arch/gaudi2/common/common.h
@@ -62,6 +62,12 @@
 #define NUM_OF_MME_SBTE_PORTS		5
 #define NUM_OF_SRAM_BANK_PER_DCORE	8
 
+/*
+ * TODO: These are temporary macros
+ * and to be removed after GAUDI2 PO
+ */
+#define PCIE_AXI_ERR_INTR_BIT		(BIT(0) | BIT(1))
+#define RAZWI_INTR_STATUS_PATTERN	0xABBA
 
 #define DCORE_OFFSET			(mmDCORE1_TPC0_QM_BASE - \
 					 mmDCORE0_TPC0_QM_BASE)
diff --git a/lib/common/cmd_module.c b/lib/common/cmd_module.c
index 0eb60a6..aaa6b48 100644
--- a/lib/common/cmd_module.c
+++ b/lib/common/cmd_module.c
@@ -12,6 +12,8 @@
 #include "cmd_module.h"
 
 __weak bool preboot_host_gpio_asserted(void) { return true; }
+/* TODO: This is temporary and to be removed after GAUDI2 PO */
+__weak void check_razwi_intr(void) {}
 
 static void i2c_debug_mode_handling(void)
 {
@@ -47,6 +49,9 @@
 	i = 0;
 	while (1) {
 
+		/* TODO: This is temporary and to be removed after GAUDI2 PO */
+		check_razwi_intr();
+
 		i2c_debug_mode_handling();
 
 		cpu_boot_if_fix_ro_regs();
